v2/pkg/engine/resolve/resolve_test.go

```diff
@@ -1876,6 +1876,64 @@ func TestResolver_ResolveGraphQLResponse(t *testing.T) {
 			},
 		}, Context{ctx: context.Background()}, `{"data":{"user":{"id":1,"name":"Jannik","__typename":"User","aliased":"User","rewritten":"User"}}}`
 	}))
+	t.Run("__typename checks", testFn(func(t *testing.T, ctrl *gomock.Controller) (node *GraphQLResponse, ctx Context, expectedOutput string) {
```

Comment:
This seems like a pretty significant change, maybe some more tests would help validate it (e.g. one that a `User` type works successfully, another test which isn't being rewritten, one where you enable multiple typenames, etc)

Reply:
All other tests validate the "correct" behaviour already.